scripts/preflight.sh

```diff
@@ -7,6 +7,27 @@ GOVERSION="$(go env GOVERSION || echo "not installed")"
 if ! [[ "$GOVERSION" == "go1.24" ||  "$GOVERSION" = "go1.24."* ]]; then
   echo "Detected go version $GOVERSION, but 1.24 is required"
   exit 1
-else
-  echo "Preflight passed ✅"
 fi
+
+if ! command -v controller-gen >/dev/null 2>&1; then
```

Comment:
should we also check that the kind is available? Or is that intentionally left off incase you want to setup your own cluster?

Reply:
I left it off since its not technically needed to build the code or generate the Docker image. Like you mentioned, we really just need a k8s cluster of some sort and kind is just one way of getting to that.